### ISSUE TITLE
Front End Users get PageMode.EDIT_MODE

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18076]
If you create a login form for use in the front end that tries to make use of the `AuthenticationResource`, it sets a default page mode to EDIT_MODE.  This has the effect of making the whole front end inaccessible to the newly logged in user who is now presented with a 403.

This is because of the `DotCustomLoginPostAction` which is fired after login here:
https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portal/struts/DotCustomLoginPostAction.java#L89

Front End Users should never have any PAGE_MODE except LIVE

**Comment by wezell** [comment]
PR: #18079

**Comment by wezell** [comment]
This PR does a number of things:

1. It unifies the variables and tooling around front-end users and back-end users.  dotCMS used to use 2 session variables to store users - `USER` and `cms.user`.  Now, only `USER` is used. To get your front end user in velocity, you would do either `$user`  or worse you could do `$session.getAttribute("USER")` 
2. A user is a user.  The only difference between a front end user and a back end user is if they have been set as "front end" or "back end" users in the admin. 
3. Front end users can only use `PageMode.LIVE`.  If a user is a front-end user, they are not allowed to specify or set a page mode that is not `LIVE`, meaning, front end users  cannot preview unpublished content, pages etc 
4. This also fixes an issue that prevented failed login attempts from being wiped out on a successful login - which would lead to long delays when logging in.

**Comment by wezell** [comment]
New Login Form

```html


<script type="text/javascript" >

   function loginDotCMS(){

      document.getElementById("messageHolder").display="none";
       let data = {};
       data.userId=document.getElementById("userId").value;
       data.password=document.getElementById("password").value;
       data.rememberMe=false;

          
      var xhr = new XMLHttpRequest();

      
      xhr.onreadystatechange = function() { 
        if(this.readyState != XMLHttpRequest.DONE){
          return;
        }

        let res = JSON.parse(this.responseText);
        if(this.status === 200) {
          console.log("login successful")
          document.getElementById("loginForm").style.display="none";

          let user = res.entity;

          let successTemplate = `
            <h2>Success</h2>
            <img src="https://www.gravatar.com/avatar/${user.gravitar}"><br>gravitar
            <hr>
            <b>Name :</b>  ${user.firstName} ${user.lastName}<br>
            <b>Email :</b> ${user.emailAddress} <br>
            <b>frontendUser :</b>  ${user.frontendUser}<br>
            <b>hasConsoleAccess :</b> ${user.hasConsoleAccess} <br>
          `;
          document.getElementById("messageHolder").style.display="block";
          document.getElementById("messageHolder").innerHTML=successTemplate;
        }else{
          console.log("login failed")
          document.getElementById("messageHolder").innerText="Invalid Login"
          document.getElementById("messageHolder").style.display="block";
          console.error("Error", this.responseText);
        }
      }

      xhr.open("POST", "/api/v1/authentication", true);
      xhr.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
      xhr.send(JSON.stringify(data));
    
    };
        


</script>

<style>
  #loginForm{
    max-width:400px;
    margin:auto;
    border:1px silver solid;
    padding:10px;
  }
  #loginForm>div{
    padding:5px;
  }
  #loginForm>div>input{
    padding:5px;
  }
  #loginForm>div>label{
    text-align:right;
    display:inline-block;
    width:120px;
  }
  #loginForm>div>button{
    padding:10px;


  }
  #messageHolder{
    max-width:300px;margin:auto;border:1px silver solid;padding:10px;margin:auto;display:none
  }


</style>

		
<div id="loginForm" >
  <h3>Login Form</h3>
	<!-- Text input-->
	<div>
		<label for="userId" >Email / UserId<span class="required">*</span></label>:
		<input id="userId"  name="userId" type="email" value="will@dotcms.com" placeholder="userId / Email" required />
	</div>

	<div>
		<label for="password">Password<span class="required">*</span></label>:
		<input id="password" name="password"  value="teste" type="password" placeholder=""/>
	</div>
	<div>
		<label for="password">Remember Me</label>:
		<input type="checkbox" id="rememberMe" name="rememberMe"  value="true" />
	</div>
	<!-- Button -->
	<div style="text-align: center;">
		<button id="loginButton" name="singlebutton" onclick="loginDotCMS()" >Login</button>
  </div>
</div>

<div id="messageHolder">

</div>

```

**Comment by fabrizzio-dotCMS** [comment]
I created a template on which I added the above script as "code". Then created a new page out of that template.  like http://localhost:8080/login/new-login
Then I followed two scenarios.  
When Login-in as Front-end user I can navigate the site with no issues whatsoever.  I was able to verify that the currently logged in user is the expected by visiting this page http://localhost:8080/login/

The front end-user has no access to the back-end which is expected.

The second scenario I tried. Login-in as Backend user. Pretty much gives me access to the backend.
